### PR TITLE
fix(docs): password visibility toggle example in form field component…

### LINKED ISSUE #15023: password visibility toggle example in form field component is not accessible
#### What is the expected behavior?
1. It must be possible to operate the password visibility toggle button in the form field example by only using a keyboard
2. Roles, states, and properties must be present so that users of screen reader software are aware that the of the toggle functionality

#### What is the current behavior?
1. It is not possible to operate the password visibility toggle button in the form field example by only using a keyboard
2. Roles, states, and properties are not present so that users of screen reader software are aware that the of the toggle functionality

#### What are the steps to reproduce?
1. Using either of the following screen readers
   - NVDA 2018.4.1 + Firefox 64
   - VoiceOver for IOS (12.1.3) + Safari
2. Open the [form field example](https://material.angular.io/components/form-field/examples)
3. Navigate to the passord visibility toggle button in the "Form field with prefix & suffix" example
   - In NVDA you can press the tab key until you reach it
   - In VoiceOver for iOS you can swipe right until you reach it
4. With NVDA observe that the tab key does not focus the button
5. With VoiceOver for iOS observe that the button is skipped
6. Observe that it is not possible to activate the button by pressing space or enter keys
7. Observe that the name, role, and state of the toggle button is not announced by the screen reader.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
- Angular Material Example 7.2.2 located at https://material.angular.io/components/
- NVDA 2018.4.1 + FireFox 64
- VoiceOver for IOS (12.1.3) + Safari (12.1.3)

#### Is there anything else we should know?
Note that this example is also included in the [form field overview page](https://material.angular.io/components/form-field/overview).

### PR DESCRIPTION
**Pull request by Suresh918**
Fixes #15023

### CODE REVIEW

File: src/material-examples/form-field-prefix-suffix/form-field-prefix-suffix-example.html [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <div class="example-container">
   <mat-form-field>
     <input matInput placeholder="Enter your password" [type]="hide ? 'password' : 'text'">
-    <mat-icon matSuffix (click)="hide = !hide">{{hide ? 'visibility_off' : 'visibility'}}</mat-icon>
+    <button mat-icon-button matSuffix (click)="hide = !hide" aria-label="Show or hide password icon">
```
Comment (crisbeto) [inline src/material-examples/form-field-prefix-suffix/form-field-prefix-suffix-example.html]: The `aria-label` show be different based on whether the button is going to show or hide the value:

```
[attr.aria-label]="hide ? 'Hide password' : 'Show password'"
````

Comment (Suresh918) [reply src/material-examples/form-field-prefix-suffix/form-field-prefix-suffix-example.html]: Changed as per your suggestion. Can you please check now

**Comment by jelbourn** [comment]
Looking at this made me realize that this should actually use a `button-toggle`. Would you be willing to change that over?

**Comment by Suresh918** [comment]
Yes sure. Will do that jelbourn

**Comment by Suresh918** [comment]
@jelbourn If we use toggle button the appearance is as shown in the image. Normal button is looking good inside the input as compared to toggle button. 
toggle button inside input

**Comment by jelbourn** [comment]
Yeah, that does look weird. I think the right thing to do here is to use a regular button with `aria-pressed` without using the whole `mat-button-toggle`

**Comment by Suresh918** [comment]
@jelbourn updated with aria-pressed

File: src/material-examples/form-field-prefix-suffix/form-field-prefix-suffix-example.html [hunk 2]
```diff
@@ -1,7 +1,9 @@
 <div class="example-container">
   <mat-form-field>
     <input matInput placeholder="Enter your password" [type]="hide ? 'password' : 'text'">
-    <mat-icon matSuffix (click)="hide = !hide">{{hide ? 'visibility_off' : 'visibility'}}</mat-icon>
+    <button mat-icon-button matSuffix (click)="hide = !hide" [attr.aria-label]="'Hide password'" [attr.aria-pressed] = hide>
```
Comment (jelbourn) [inline src/material-examples/form-field-prefix-suffix/form-field-prefix-suffix-example.html]: Small nit: omit spaces and add double-quotes:
```html
[attr.aria-pressed]="hide"
```

**Pull request merged**